conans/paths/simple_paths.py

```diff
@@ -0,0 +1,109 @@
+
+    def _build_path_to_linked_folder_sentinel(self, conan_reference):
+        base_folder = self._build_path_to_base_folder(conan_reference)
+        linked_package_file = os.path.join(base_folder, LINKED_FOLDER_SENTINEL)
+        return linked_package_file
+
+    def package_layout(self, conan_reference, short_paths=False):
+        assert isinstance(conan_reference, ConanFileReference), \
+            "It is a {}".format(type(conan_reference))
+        linked_package_file = self._build_path_to_linked_folder_sentinel(conan_reference)
+        if os.path.exists(linked_package_file):
+            return PackageEditableLayout(linked_package_file=linked_package_file,
+                                         conan_ref=conan_reference)
+        else:
+            check_ref_case(conan_reference, self.store)
+            base_folder = self._build_path_to_base_folder(conan_reference)
+            return PackageCacheLayout(base_folder=base_folder,
+                                      conan_ref=conan_reference, short_paths=short_paths)
+
+    def conan(self, conan_reference):
+        """ the base folder for this package reference, for each ConanFileReference
+        """
+        return self.package_layout(conan_reference).conan()
+
+    def export(self, conan_reference):
+        return self.package_layout(conan_reference).export()
+
+    def export_sources(self, conan_reference, short_paths=False):
+        return self.package_layout(conan_reference, short_paths).export_sources()
+
+    def source(self, conan_reference, short_paths=False):
+        return self.package_layout(conan_reference, short_paths).source()
+
+    def conanfile(self, conan_reference):
+        return self.package_layout(conan_reference).conanfile()
+
+    def builds(self, conan_reference):
+        return self.package_layout(conan_reference).builds()
+
+    def build(self, package_reference, short_paths=False):
+        return self.package_layout(package_reference.conan, short_paths).build(package_reference)
+
+    def system_reqs(self, conan_reference):
+        return self.package_layout(conan_reference).system_reqs()
+
+    def system_reqs_package(self, package_reference):
+        return self.package_layout(package_reference.conan).system_reqs_package(package_reference)
+
+    def packages(self, conan_reference):
+        return self.package_layout(conan_reference).packages()
+
+    def package(self, package_reference, short_paths=False):
+        return self.package_layout(package_reference.conan, short_paths).package(package_reference)
+
+    def scm_folder(self, conan_reference):
+        return self.package_layout(conan_reference).scm_folder()
+
+    def package_metadata(self, conan_reference):
+        return self.package_layout(conan_reference).package_metadata()
+
+    def installed_as_editable(self, conan_reference):
+        return self.package_layout(conan_reference).installed_as_editable()
```

Comment:
Check the sentinel directly instead.

Reply:
It is the responsibility of the package_layout class to answer if it is editable or not.

At this moment it is a matter of the sentinel being there or not, but tomorrow we might need to load the sentinel and evaluate its content to know if it is pointing to a valid folder or not.

Reply:
A package layout cannot be instantiated at all without knowing if it is editable or not. So the responsibility to know when a pkg-reference is editable or not, does not belong to the layout.

Reply:
You can check the uses of the ``PackageLayout.installed_as_editable()``. It is only used when called by the ``ClientCache.installed_as_editable()``, that calls itself in the ``self.package_layout()`` method, to check if it has the sentinel or not to define a package as editable.

Reply:
I can change this for a `return isinstance(self.package_layout(conan_reference), PackageEditableLayout)`, but I'm not going to duplicate the code that checks for the sentinel. 
Nevertheless, I prefer to keep this function. If you want me to, I can do the following somewhere in the sources (and it will be much better/cleaner than passing the `client_cache` everywhere):

```python
...
package_layout = self.client_cache.package_layour(reference)
...

if package_layout.installed_as_editable():
    ...
```

Reply:
Ok, good, go for the current implementation